modules/ttdBidAdapter.js

```diff
@@ -0,0 +1,602 @@
+      return false;
+    }
+    if (bid.params.publisherId.length > 32) {
+      utils.logWarn(BIDDER_CODE + ': params.publisherId must be 32 characters or less');
+      return false;
+    }
+    if (!bid.params.siteId) {
+      utils.logWarn(BIDDER_CODE + ': Missing required parameter params.siteId');
+      return false;
+    }
+    if (bid.params.siteId.length > 50) {
+      utils.logWarn(BIDDER_CODE + ': params.siteId must be 50 characters or less');
+      return false;
+    }
+    if (!bid.params.placementId) {
+      utils.logWarn(BIDDER_CODE + ': Missing required parameter params.placementId');
+      return false;
+    }
+    if (bid.params.placementId.length > 128) {
+      utils.logWarn(BIDDER_CODE + ': params.placementId must be 128 characters or less');
+      return false;
+    }
+    if (bid.params.secure && !(bid.params.secure === 0 || bid.params.secure === 1)) {
+      utils.logWarn(BIDDER_CODE + ': params.secure should be either 0 or 1');
+      return false;
+    }
+    const mediaTypesVideo = utils.deepAccess(bid, 'mediaTypes.video');
+    const mediaTypesBanner = utils.deepAccess(bid, 'mediaTypes.banner');
+
+    if (mediaTypesBanner || !(mediaTypesVideo || mediaTypesBanner)) {
+      const bannerExpDir = utils.deepAccess(bid, 'params.banner.expdir');
+      if (bannerExpDir) {
+        if (!Array.isArray(bannerExpDir)) {
+          utils.logWarn(BIDDER_CODE + ': params.banner.expdir should be an array');
+          return false;
+        }
+      }
+    } else {
+      if (!mediaTypesVideo.minduration || !utils.isInteger(mediaTypesVideo.minduration)) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.minduration must be set to the minimum video ad duration in seconds');
+        return false;
+      }
+      if (!mediaTypesVideo.maxduration || !utils.isInteger(mediaTypesVideo.maxduration)) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.maxduration must be set to the maximum video ad duration in seconds');
+        return false;
+      }
+      if (!mediaTypesVideo.playerSize) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.playerSize should be an array of width, height values')
+        return false;
+      }
+      if (!mediaTypesVideo.api || mediaTypesVideo.api.length === 0) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.api should be an array of supported api frameworks. See the Open RTB v2.5 spec for valid values');
+        return false;
+      }
+      if (!mediaTypesVideo.mimes || mediaTypesVideo.mimes.length === 0) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.mimes should be an array of supported mime types');
+        return false;
+      }
+      if (!mediaTypesVideo.protocols) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.protocols should be an array of supported protocols. See the Open RTB v2.5 spec for valid values')
+        return false;
+      }
+      if (mediaTypesVideo.startdelay && !utils.isInteger(mediaTypesVideo.startdelay)) {
```

Comment:
If these are not required params, why reject the bid completely here and below? I feel like this is not a beneficial use of conditionals here. You can simply use your conditions within the video method to determine whether or not to set these optional params. i.e. instead of rejecting startdelay when it is set but not an integer, you can move this logic up to the video method and run something similar to this: 

```
if (startdelay && utils.isInteger(startdelay)) {
    video.startdelay = startdelay;
  }
```

This way you can drop the optional properties deemed invalid without rejecting the entire bid. This will also reduce the number of conditionals you have here

Reply:
Thank you, I've updated the code to use your pattern of adding these optional params only if they're valid without rejecting the bid.